### ISSUE TITLE
UnicodeEncodeError: 'ascii' codec can't encode character '\u2018' in position 11: ordinal not in range(128)

### ISSUE THREAD

**Issue by tabka** [mathiasertl/fabric#28]
Hello I have always the same error when I try to run a wget command, the issue seems to be present only in fabric3. I can't manage to get the output of the command it always raises this error when I run my program or even on ipython:
`UnicodeEncodeError: 'ascii' codec can't encode character '\u2018' in position 11: ordinal not in range(128)`

Here's the output of ipython:

```
In [9]: out = sudo('wget -O /tmp/CentOS-7-x86_64-Minimal-1611.iso http://it.centos.contactlab.it/7/isos/x86_64/CentOS-7-x86_64-Minimal-1611.iso')
[192.168.1.2] sudo: wget -O /tmp/CentOS-7-x86_64-Minimal-1611.iso http://it.centos.contactlab.it/7/isos/x86_64/CentOS-7-x86_64-Minimal-1611.iso
[192.168.1.2] out: --2017-02-20 14:08:26--  http://it.centos.contactlab.it/7/isos/x86_64/CentOS-7-x86_64-Minimal-1611.iso
[192.168.1.2] out: Resolving it.centos.contactlab.it (it.centos.contactlab.it)... 185.34.86.124
[192.168.1.2] out: Connecting to it.centos.contactlab.it (it.centos.contactlab.it)|185.34.86.124|:80... connected.
[192.168.1.2] out: HTTP request sent, awaiting response... 200 OK
[192.168.1.2] out: Length: 713031680 (680M) [application/x-iso9660-image]
[192.168.1.2] out: ---------------------------------------------------------------------------
UnicodeEncodeError                        Traceback (most recent call last)
<ipython-input-9-bc8569c418dc> in <module>()
----> 1 out = sudo('wget -O /tmp/CentOS-7-x86_64-Minimal-1611.iso http://it.centos.contactlab.it/7/isos/x86_64/CentOS-7-x86_64-Minimal-1611.iso')

/home/.virtualenvs/core/lib/python3.4/site-packages/fabric/network.py in host_prompting_wrapper(*args, **kwargs)
    654                 host_string = raw_input(prompt)
    655             env.update(to_dict(host_string))
--> 656         return func(*args, **kwargs)
    657     host_prompting_wrapper.undecorated = func
    658     return host_prompting_wrapper

/home.virtualenvs/core/lib/python3.4/site-packages/fabric/operations.py in sudo(command, shell, pty, combine_stderr, user, quiet, warn_only, stdout, stderr, group, timeout, shell_escape)
   1121         user=user if user else env.sudo_user,
   1122         group=group, quiet=quiet, warn_only=warn_only, stdout=stdout,
-> 1123         stderr=stderr, timeout=timeout, shell_escape=shell_escape,
   1124     )
   1125 

/home/virtualenvs/core/lib/python3.4/site-packages/fabric/operations.py in _run_command(command, shell, pty, combine_stderr, sudo, user, quiet, warn_only, stdout, stderr, group, timeout, shell_escape)
    931             channel=default_channel(), command=wrapped_command, pty=pty,
    932             combine_stderr=combine_stderr, invoke_shell=False, stdout=stdout,
--> 933             stderr=stderr, timeout=timeout)
    934 
    935         # Assemble output string

/home/.virtualenvs/core/lib/python3.4/site-packages/fabric/operations.py in _execute(channel, command, pty, combine_stderr, invoke_shell, stdout, stderr, timeout)
    815                 # exception within, recv_exit_status())
    816                 for worker in workers:
--> 817                     worker.raise_if_needed()
    818             try:
    819                 time.sleep(ssh.io_sleep)

/home/.virtualenvs/core/lib/python3.4/site-packages/fabric/thread_handling.py in raise_if_needed(self)
     24         if self.exception:
     25             e = self.exception
---> 26             six.reraise(e[0], e[1], e[2])

/home/.virtualenvs/core/lib/python3.4/site-packages/six.py in reraise(tp, value, tb)
    609         if value.__traceback__ is not tb:
    610             raise value.with_traceback(tb)
--> 611         raise value
    612 
    613 else:

/home/.virtualenvs/core/lib/python3.4/site-packages/fabric/thread_handling.py in wrapper(*args, **kwargs)
     11         def wrapper(*args, **kwargs):
     12             try:
---> 13                 callable(*args, **kwargs)
     14             except BaseException:
     15                 self.exception = sys.exc_info()

/home/.virtualenvs/core/lib/python3.4/site-packages/fabric/io.py in output_loop(*args, **kwargs)
     31 
     32 def output_loop(*args, **kwargs):
---> 33     OutputLooper(*args, **kwargs).loop()
     34 
     35 

/home/.virtualenvs/core/lib/python3.4/site-packages/fabric/io.py in loop(self)
    155                             line = []
    156                         else:
--> 157                             self._flush(end_of_line + "\n")
    158                         initial_prefix_printed = False
    159 

/home/.virtualenvs/core/lib/python3.4/site-packages/fabric/io.py in _flush(self, text)
     52 
     53     def _flush(self, text):
---> 54         self.stream.write(text)
     55         # Actually only flush if not in linewise mode.
     56         # When linewise is set (e.g. in parallel mode) flushing makes

UnicodeEncodeError: 'ascii' codec can't encode character '\u2018' in position 11: ordinal not in range(128)
```

Edit: when I run the wget command with --quiet option the exception is not raised so there's something in the output of the command , the result is not encoded properly.

 
Thank you for your help!

**Comment by mathiasertl** [comment]
I cannot reproduce this on my machine. I tried this task:

```
@task
def test_28():
    path = '/tmp/test.iso'
    url = 'http://it.centos.contactlab.it/7/isos/x86_64/CentOS-7-x86_64-Minimal-1611.iso'  # NOQA
    local('wget -O %s %s' % (path, url))
    local('rm %s' % path)

    run('wget -O %s %s' % (path, url))
    run('rm %s' % path)

    sudo('wget -O %s %s' % (path, url))
    sudo('rm %s' % path)
```

This works for me. Can you give me more info about your environment? What python version? What OS and Version? Version of wget?

**Comment by tabka** [comment]
I made the tests and now its not happening again, I'm not sure what occurred, but I was stuck with this issue for days and now it's gone.
The most important part that everything is now working, sorry for the disturbance.

**Comment by mathiasertl** [comment]
Ok, I'm closing this issue for now. If you encounter this again, please don't hesitate to reopen it or open a new issue! Be sure to include as much info as possible to help me reproduce the error ;-)

**Comment by KyeRussell** [comment]
I'm having what I believe to be the same issue.

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/fabric/main.py", line 762, in main
    *args, **kwargs
  File "/usr/local/lib/python3.5/dist-packages/fabric/tasks.py", line 385, in execute
    multiprocessing
  File "/usr/local/lib/python3.5/dist-packages/fabric/tasks.py", line 275, in _execute
    return task.run(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/fabric/tasks.py", line 172, in run
    return self.wrapped(*args, **kwargs)
  File "/opt/project/fabfile/aws/ec2.py", line 96, in setup
    dist_upgrade()
  File "/usr/local/lib/python3.5/dist-packages/fabric/tasks.py", line 169, in __call__
    return self.run(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/fabric/tasks.py", line 172, in run
    return self.wrapped(*args, **kwargs)
  File "/opt/project/fabfile/aws/ec2.py", line 161, in dist_upgrade
    sudo('apt-get dist-upgrade --quiet --yes')
  File "/usr/local/lib/python3.5/dist-packages/fabric/network.py", line 682, in host_prompting_wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/fabric/operations.py", line 1149, in sudo
    capture_buffer_size=capture_buffer_size,
  File "/usr/local/lib/python3.5/dist-packages/fabric/operations.py", line 934, in _run_command
    capture_buffer_size=capture_buffer_size)
  File "/usr/local/lib/python3.5/dist-packages/fabric/operations.py", line 816, in _execute
    worker.raise_if_needed()
  File "/usr/local/lib/python3.5/dist-packages/fabric/thread_handling.py", line 26, in raise_if_needed
    six.reraise(e[0], e[1], e[2])
  File "/usr/local/lib/python3.5/dist-packages/six.py", line 686, in reraise
    raise value
  File "/usr/local/lib/python3.5/dist-packages/fabric/thread_handling.py", line 13, in wrapper
    callable(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/fabric/io.py", line 31, in output_loop
    OutputLooper(*args, **kwargs).loop()
  File "/usr/local/lib/python3.5/dist-packages/fabric/io.py", line 161, in loop
    self._flush(printable_bytes)
  File "/usr/local/lib/python3.5/dist-packages/fabric/io.py", line 52, in _flush
    self.stream.write(text)
UnicodeEncodeError: 'ascii' codec can't encode characters in position 27-71: ordinal not in range(128)
Disconnecting from ec2-13-211-47-129.ap-southeast-2.compute.amazonaws.com... done.
```

Are you able to provide any input?
